# 4장. 파이썬 기초 

## 파이썬을 계산기로 사용해보자.


수식을 입력해보자. 

In [1]:
1+1+1

3

In [2]:
(1+1+1)/3

1.0

In [4]:
1*2*3*4*5

120

In [3]:
10+20+40

70

메뉴에서 'Insert Cell Below' 를 선택해서 셀을 추가한 다음에 수식을 넣고, 실행을 해보자.

## 변수 (Variable) 

- 변수는 '무엇인가' 를 저장하는 저장소라고 생각하면 된다. 
- '무엇인가' 는 숫자, 글자, 기타 데이터 등 여러가지가 될 수 있음. 

In [5]:
a=1
b=1
c = a+b

a의 값을 살펴보고 싶으면...

In [11]:
a=3
a

3

b의 값을 살펴보고 싶으면...

In [7]:
b

1

In [14]:
c

2

글자를 저장하는 것도 가능하다. 컴퓨터 언에서 보통 글자는 '문자열' 혹은 '스트링' (String) 이라고 부른다. 
글자는 작은 따옴표 혹은 큰 따옴표로 구분한다. 

In [16]:
d = "Python"
e = " is good" 
f = d + e 

In [22]:
print (f)

Python is good


In [24]:
a="ACGATAGAGAGG"

In [26]:
print (a)

ACGATAGAGAGG


In [ ]:
'='는 '같다'가 아니라 변수에 데이터를 넣는 기호이다. 
a = 'this is good' 

a에 'this is good'을 넣는다. 

In [13]:
a='this is good'
a

'this is good'

'문자열' 은 서열 정보를 저장하는데도 사용할 수 있다. 
다음 예를 본다면...

In [27]:
sequence = "ACGTACGTAG"
print (sequence)

ACGTACGTAG


문자열끼리 '더하기' 를 하면 두 개의 서열이 이어진다. 

In [28]:
sequence2 = "GGATCCGT"
new_sequence = sequence + sequence2
print (new_sequence)

ACGTACGTAGGGATCCGT


In [44]:
new_sequence[18]

IndexError: string index out of range

만약 이들을 각각의 위치별의 글자로 읽으려면 어떻게 해야 하나? 
문자열의 일부에 접근하려면 대괄호 '[]' 와 숫자를 이용한다. 

In [17]:
new_sequence[0]

'A'

In [18]:
new_sequence[1]

'C'

대괄호 안의 숫자를 바꾸어 보자. 
첫번째 항목은 0 으로부터 시작한다. 

여러 개의 항목을 접근하려면 [시작위치:끝위치] 형식으로 숫자를 이용한다. 

In [46]:
new_sequence[0:3]

'ACG'

In [47]:
new_sequence[0:10]

'ACGTACGTAG'

만약 세번째부터 끝까지의 서열에 접근하고 싶다면? 

In [48]:
#0,1,2....
new_sequence[2:]

'GTACGTAGGGATCCGT'

세번째까지의 서열을 알고 싶다면...

In [50]:
new_sequence[:4]

'ACGT'

In [55]:
#끝에서 3글자 빼기

new_sequence[:-3]

'ACGTACGTAGGGATC'

In [56]:
new_sequence[3:5]

'TA'

## 리스트 (LIST) 

- 여러 개의 데이터를 저장할 때 사용한다. 
- 대괄호와 쉼표로 여러 개의 데이터를 구분하여 저장한다. 
- 숫자, 쉼표, 다른 변수 등등 여러 종류의 데이터를 저장 가능하다. 


In [57]:
even = [2,4,6,8,10,12]

In [59]:
odd = [1,3,5,7,9]

In [26]:
even = [2,4,6,8,10,12]
odd = [1,3,5,7,9]
even

[2, 4, 6, 8, 10, 12]

In [62]:
string = 'this is good'
string

'this is good'

In [63]:
string_list = ['one', 'two','three','four','five']
string_list

['one', 'two', 'three', 'four', 'five']

리스트의 항목도 문자열 안의 글자와 비슷하게 참조할 수 있다. 

In [66]:
string_list[4]

'five'

In [31]:
string_list[4]

'five'

In [68]:
odd[1]

3

In [ ]:
리스트의 항목 일부도 마찬가지로 억세스할 수 있다. 
string_list의 첫번째에서 세번째 항목까지 억세스하려면? 

In [71]:
string_list[0:3]

['one', 'two', 'three']

두 개의 리스트를 더할 수도 있다. 

[1, 3, 5, 7, 9]

In [74]:
number = even + odd

In [77]:
number.append(100)

하나를 추가하고 싶다고 하자. '100'

In [37]:
number.append(100)

In [80]:
string_list.append('hundred')

In [82]:
string_list.append('hundred one')

In [83]:
string_list

['one', 'two', 'three', 'four', 'five', 'hundred', 'hundred one']

In [39]:
string_list.append('six')

In [40]:
string_list

['one', 'two', 'three', 'four', 'five', 'six']

In [85]:
number.sort()

number라는 리스트에 있는 내용을 숫자별로 정렬하고 싶다면? 


In [86]:
number

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 100]

In [43]:
number

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 12, 100]

In [88]:
string_list.append('aaaaaa')

In [91]:
string_list

['aaaaaa', 'five', 'four', 'hundred', 'hundred one', 'one', 'three', 'two']

In [46]:
string_list.sort()
string_list

['aaaaaa', 'five', 'four', 'one', 'six', 'three', 'two']

리스트에 있는 내용을 바꾸고 싶다면? 
가령 두번째의 'five'를 'FIVE'로 바꾸고 싶다고 하자. 


In [47]:
string_list[1]

'five'

In [48]:
string_list[1]='FIVE'

In [49]:
string_list

['aaaaaa', 'FIVE', 'four', 'one', 'six', 'three', 'two']

In [50]:
del string_list[1]

'AAGGAGAGG'

## 딕셔너리 (Dictionary)  

- 서로 관계가 있는 데이터를 저장하고 싶다고 하자. 
- Elephant, Animal
- Sunflower, Plant
- E.coli, Microbe
- Human, Animal
- Rice, Plant 

- 이러한 관계를 어떻게 저장할 것인가? 

In [96]:
organism = {'Elephant':'Animal',
            'Sunflower':'Plant',
            'E.coli':'Microbe',
            'Human':'Animal',
            'Rice':'Plant'}

딕셔너리에서 앞의 내용을 '키' (Key)라고 하고 뒤의 내용을 '값' (Value)라고 한다. 
하나의 딕셔너리에서는 키의 값은 고유한 값이어야 한다. 

In [99]:
organism['Rice']

'Plant'

In [55]:
#키를 이용하여 값을 알아낸다. 

organism['E.coli']

'Microbe'

In [101]:
organism['Dog']='Animal'
organism

{'Elephant': 'Animal',
 'Sunflower': 'Plant',
 'E.coli': 'Microbe',
 'Human': 'Animal',
 'Rice': 'Plant',
 'Dog': 'Animal'}

딕셔너리에 새로운 데이터를 넣고 싶을 경우
딕셔너리['키'] = '값' 

In [103]:
organism['Dog']="Human's best friend"

In [104]:
organism['Dog']

"Human's best friend"

In [109]:
organism['Dog']='Good Food'
organism['Dog']='Do not eat dog'
organism

{'Elephant': 'Animal',
 'Sunflower': 'Plant',
 'E.coli': 'Microbe',
 'Human': 'Animal',
 'Rice': 'Plant',
 'Dog': 'Do not eat dog'}

딕셔너리에는 동일한 키 값이 두 개 이상 존재할 수 없다. 

In [110]:
organism.keys()

dict_keys(['Elephant', 'Sunflower', 'E.coli', 'Human', 'Rice', 'Dog'])

In [111]:
organism.values()

dict_values(['Animal', 'Plant', 'Microbe', 'Animal', 'Plant', 'Do not eat dog'])

In [63]:
organism.values()

dict_values(['Animal', 'Plant', 'Microbe', 'Animal', 'Plant', 'Humans best friends'])

## 튜플 (Tuple)  

- 튜플은 리스트와 거의 비슷하나 하나의 차이가 있다. 

In [112]:
list = [1,2,3,4]
tuple = (1,2,3,4)

In [113]:
list

[1, 2, 3, 4]

In [114]:
tuple

(1, 2, 3, 4)

In [115]:
list[0] = 2
list

[2, 2, 3, 4]

In [117]:
tuple[0]=1000

TypeError: 'tuple' object does not support item assignment

리스트는 값을 추가하거나 바꿀 수 있지만, 튜플은 바꿀 수 없다. 
튜플은 한번 넣어두면 바뀌지 않는 값을 여러 개 저장할 때 사용한다. 

In [69]:
base = ('A','C','G','T')

## 지금까지 파이썬에서 어떻게 자료를 저장하는지 배웠음. 

- 변수 
- 리스트 
- 딕셔너리
- 튜플
- 문자열 

## 조건과 반복

컴퓨터 프로그래밍의 가장 기본적인 두 가지 요소 

- 반복 
- 조건 판단 

리스트에 들어 있는 여러 값을 하나씩 꺼내서 살펴보는 과정을 보자. 

In [119]:
test_list = ['권은비', 
             '미야와키 사쿠라', 
             '강혜원', 
             '최예나', 
             '이채연', 
             '김채원', 
             '김민주', 
             '야부키 나코', 
             '혼다 히토미', 
             '조유리', 
             '안유진', 
             '장원영']

In [127]:
for member in test_list:
    print (member)
    print ("....")

IndentationError: unexpected indent (<ipython-input-127-6eb78c3d463e>, line 3)

파이썬에서는 반복문이나 조건문 내에서 수행되는 라인을 구분하기 위하여 '들여쓰기' 를 한다. 
뭔 말이냐면...다음과 같은 코드를 입력하면..

In [11]:
#에러 발생
for member in test_list:
print (member)

IndentationError: expected an indented block (<ipython-input-11-044f31795d4e>, line 2)

In [14]:
#수정됨
for member in test_list:
    print (member)

권은비
미야와키 사쿠라
강혜원
최예나
이채연
김채원
김민주
야부키 나코
혼다 히토미
조유리
안유진
장원영


In [15]:
for member in test_list:
    #반복문 안의 문장들은 들여쓰기 (Tab) 레벨이 같아야 한다. 
    print (member)
        print ("..")

IndentationError: unexpected indent (<ipython-input-15-1c328a6c1689>, line 4)

# range 

일정한 숫자를 이용하여 반복을 한다. 

In [129]:
for i in range(100):
    print (i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99


In [131]:
for i in range(1,10):
    for j in range(1,10):
        print (i,j)

1 1
1 2
1 3
1 4
1 5
1 6
1 7
1 8
1 9
2 1
2 2
2 3
2 4
2 5
2 6
2 7
2 8
2 9
3 1
3 2
3 3
3 4
3 5
3 6
3 7
3 8
3 9
4 1
4 2
4 3
4 4
4 5
4 6
4 7
4 8
4 9
5 1
5 2
5 3
5 4
5 5
5 6
5 7
5 8
5 9
6 1
6 2
6 3
6 4
6 5
6 6
6 7
6 8
6 9
7 1
7 2
7 3
7 4
7 5
7 6
7 7
7 8
7 9
8 1
8 2
8 3
8 4
8 5
8 6
8 7
8 8
8 9
9 1
9 2
9 3
9 4
9 5
9 6
9 7
9 8
9 9


for 문은 중첩 가능하다

In [132]:
for x in range (1,10):
    for y in range(1,10):
        print (x,y,x*y)

1 1 1
1 2 2
1 3 3
1 4 4
1 5 5
1 6 6
1 7 7
1 8 8
1 9 9
2 1 2
2 2 4
2 3 6
2 4 8
2 5 10
2 6 12
2 7 14
2 8 16
2 9 18
3 1 3
3 2 6
3 3 9
3 4 12
3 5 15
3 6 18
3 7 21
3 8 24
3 9 27
4 1 4
4 2 8
4 3 12
4 4 16
4 5 20
4 6 24
4 7 28
4 8 32
4 9 36
5 1 5
5 2 10
5 3 15
5 4 20
5 5 25
5 6 30
5 7 35
5 8 40
5 9 45
6 1 6
6 2 12
6 3 18
6 4 24
6 5 30
6 6 36
6 7 42
6 8 48
6 9 54
7 1 7
7 2 14
7 3 21
7 4 28
7 5 35
7 6 42
7 7 49
7 8 56
7 9 63
8 1 8
8 2 16
8 3 24
8 4 32
8 5 40
8 6 48
8 7 56
8 8 64
8 9 72
9 1 9
9 2 18
9 3 27
9 4 36
9 5 45
9 6 54
9 7 63
9 8 72
9 9 81


구구단?

In [75]:
for x in range (1,10):
    for y in range(1,10):
        print (x,y,x*y)

1 1 1
1 2 2
1 3 3
1 4 4
1 5 5
1 6 6
1 7 7
1 8 8
1 9 9
2 1 2
2 2 4
2 3 6
2 4 8
2 5 10
2 6 12
2 7 14
2 8 16
2 9 18
3 1 3
3 2 6
3 3 9
3 4 12
3 5 15
3 6 18
3 7 21
3 8 24
3 9 27
4 1 4
4 2 8
4 3 12
4 4 16
4 5 20
4 6 24
4 7 28
4 8 32
4 9 36
5 1 5
5 2 10
5 3 15
5 4 20
5 5 25
5 6 30
5 7 35
5 8 40
5 9 45
6 1 6
6 2 12
6 3 18
6 4 24
6 5 30
6 6 36
6 7 42
6 8 48
6 9 54
7 1 7
7 2 14
7 3 21
7 4 28
7 5 35
7 6 42
7 7 49
7 8 56
7 9 63
8 1 8
8 2 16
8 3 24
8 4 32
8 5 40
8 6 48
8 7 56
8 8 64
8 9 72
9 1 9
9 2 18
9 3 27
9 4 36
9 5 45
9 6 54
9 7 63
9 8 72
9 9 81


1부터 5까지만 계산하려면 어떻게 코드를 고쳐야 할까? 

In [135]:
for x in range (1,10):
    for y in range(1,5):
        print (x)
        print (x,y,x*y)

1
1 1 1
1
1 2 2
1
1 3 3
1
1 4 4
2
2 1 2
2
2 2 4
2
2 3 6
2
2 4 8
3
3 1 3
3
3 2 6
3
3 3 9
3
3 4 12
4
4 1 4
4
4 2 8
4
4 3 12
4
4 4 16
5
5 1 5
5
5 2 10
5
5 3 15
5
5 4 20
6
6 1 6
6
6 2 12
6
6 3 18
6
6 4 24
7
7 1 7
7
7 2 14
7
7 3 21
7
7 4 28
8
8 1 8
8
8 2 16
8
8 3 24
8
8 4 32
9
9 1 9
9
9 2 18
9
9 3 27
9
9 4 36


## 조건문 if 

컴퓨터 언어의 기본은 판단 
구구단의 결과 중에서 50 보다 큰 값만 출력하고 싶다면? 


In [141]:
for x in range (1,10):
    for y in range(1,10):
        #만약 x와 y를 곱한 값이 50보다 크면 
        if x != 2:
            #x,y, 그리고 x*y 값을 출력해라 
            print (x,y,x*y)

1 1 1
1 2 2
1 3 3
1 4 4
1 5 5
1 6 6
1 7 7
1 8 8
1 9 9
3 1 3
3 2 6
3 3 9
3 4 12
3 5 15
3 6 18
3 7 21
3 8 24
3 9 27
4 1 4
4 2 8
4 3 12
4 4 16
4 5 20
4 6 24
4 7 28
4 8 32
4 9 36
5 1 5
5 2 10
5 3 15
5 4 20
5 5 25
5 6 30
5 7 35
5 8 40
5 9 45
6 1 6
6 2 12
6 3 18
6 4 24
6 5 30
6 6 36
6 7 42
6 8 48
6 9 54
7 1 7
7 2 14
7 3 21
7 4 28
7 5 35
7 6 42
7 7 49
7 8 56
7 9 63
8 1 8
8 2 16
8 3 24
8 4 32
8 5 40
8 6 48
8 7 56
8 8 64
8 9 72
9 1 9
9 2 18
9 3 27
9 4 36
9 5 45
9 6 54
9 7 63
9 8 72
9 9 81


#조건문과 딕셔너리를 결합 

In [142]:
#모 걸그룹의 멤버와 출생년도를 딕셔너리 형태로 저장한다. 
izone = {'권은비':1995, 
         '미야와키 사쿠라':1998, 
         '강혜원':1999, 
         '최예나':1999, 
         '이채연':2000, 
         '김채원':2000, 
         '김민주':2001, 
         '야부키 나코':2001, 
         '혼다 히토미':2001, 
         '조유리':2001, 
         '안유진':2003, 
         '장원영':2004}

#딕셔너리의 키와 값들을 하나씩 훑어본다. 
for key,value in izone.items():
    print (key,value)

권은비 1995
미야와키 사쿠라 1998
강혜원 1999
최예나 1999
이채연 2000
김채원 2000
김민주 2001
야부키 나코 2001
혼다 히토미 2001
조유리 2001
안유진 2003
장원영 2004


2000년생 이전 멤버만 출력하고 싶다면?

In [145]:
for key,value in izone.items():
    if value>=2000:
        print (key,value)

이채연 2000
김채원 2000
김민주 2001
야부키 나코 2001
혼다 히토미 2001
조유리 2001
안유진 2003
장원영 2004


'야부키 나코' 보다 동생만 출력하고 싶다면?

In [149]:
#nako라는 변수에 '야부키 나코'의 나이를 저장하고 

nako=izone['야부키 나코']
nako

2001

In [154]:
for key,value in izone.items():
    if nako!=value:
        print (key,value)

권은비 1995
미야와키 사쿠라 1998
강혜원 1999
최예나 1999
이채연 2000
김채원 2000
안유진 2003
장원영 2004


## if..else

만약...이면 이걸 하고 아니면 저걸 한다. 

In [155]:
for key,value in izone.items():
    if nako<value:
        print (key+" 동생")
    else:
        print (key+" 동갑 아니면 언니")

권은비 동갑 아니면 언니
미야와키 사쿠라 동갑 아니면 언니
강혜원 동갑 아니면 언니
최예나 동갑 아니면 언니
이채연 동갑 아니면 언니
김채원 동갑 아니면 언니
김민주 동갑 아니면 언니
야부키 나코 동갑 아니면 언니
혼다 히토미 동갑 아니면 언니
조유리 동갑 아니면 언니
안유진 동생
장원영 동생


만약 조건 A면 (if) A를 하고 만약 조건 B면 (elif) B를 하고..이도저도 아니면 (else) C를 한다 

In [156]:
for key,value in izone.items():
    if nako<value:
        print (key+" 동생")
    elif nako==value:
        print (key+" 동갑")
    else:
        print (key+" 언니")

권은비 언니
미야와키 사쿠라 언니
강혜원 언니
최예나 언니
이채연 언니
김채원 언니
김민주 동갑
야부키 나코 동갑
혼다 히토미 동갑
조유리 동갑
안유진 동생
장원영 동생


## 연습문제 

'미야와키 사쿠라' (혹은 아무 멤버와) 다른 모든 멤버와의 나이를 출력해보자. 

In [159]:
#아래 스크립트를 고쳐서...

izone = {'권은비':1995, 
         '미야와키 사쿠라':1998, 
         '강혜원':1999, 
         '최예나':1999, 
         '이채연':2000, 
         '김채원':2000, 
         '김민주':2001, 
         '야부키 나코':2001, 
         '혼다 히토미':2001, 
         '조유리':2001, 
         '안유진':2003, 
         '장원영':2004}

#미야와키 사쿠라의 이름을 'OnePick'이라는 변수에 저장하고 
OnePick = '미야와키 사쿠라'

#미야와키 사쿠라의 나이를 딕셔너리에서 불러온다. 
OnePickYear = izone[OnePick]


    

In [162]:
for key,value in izone.items():
    print (key, OnePickYear-value)
    #여기서 다음 출력이 나오려면 어떻게 입력해야 할까? 

권은비 3
미야와키 사쿠라 0
강혜원 -1
최예나 -1
이채연 -2
김채원 -2
김민주 -3
야부키 나코 -3
혼다 히토미 -3
조유리 -3
안유진 -5
장원영 -6


덕질은 그만하고...
이제 좀 연구에 가까운 것을 해보자 (...)

In [163]:
seq="""MEEPQSDPSVEPPLSQETFSDLWKLLPENNVLSPLPSQAMDDLMLSPDDIEQWFTEDPGPDEAPRMPEAAPPVAPAPAAPTPAAPAPAPSWPLSSSVPSQKTYQGSYGFRLGFLHSGTAKSVTCTYSPALNKMFCQLAKTCPVQLWVDSTPPPGTRVRAMAIYKQSQHMTEVVRRCPHHERCSDSDGLAPPQHLIRVEGNLRVEYLDDRNTFRHSVVVPYEPPEVGSDCTTIHYNYMCNSSCMGGMNRRPILTIITLEDSSGNLLGRNSFEVRVCACPGRDRRTEEENLRKKGEPHHELPPGSTKRALPNNTSSSPQPKKKPLDGEYFTLQIRGRERFEMFRELNEALELKDAQAGKEPGGSRAHSSHLKSKKGQSTSRHKKLMFKTEGPDSD"""

In [165]:
len(seq)

393

이 아미노산의 길이는?

In [171]:
MolecularWeight = 0
for aa in seq:
    print (aa)
    print (MW[aa])
    MolecularWeight = MolecularWeight + MW[aa]

M
131.19
E
129.12
E
129.12
P
97.12
Q
128.14
S
87.08
D
115.09
P
97.12
S
87.08
V
99.14
E
129.12
P
97.12
P
97.12
L
113.16
S
87.08
Q
128.14
E
129.12
T
101.11
F
147.18
S
87.08
D
115.09
L
113.16
W
186.21
K
128.17
L
113.16
L
113.16
P
97.12
E
129.12
N
114.11
N
114.11
V
99.14
L
113.16
S
87.08
P
97.12
L
113.16
P
97.12
S
87.08
Q
128.14
A
71.09
M
131.19
D
115.09
D
115.09
L
113.16
M
131.19
L
113.16
S
87.08
P
97.12
D
115.09
D
115.09
I
113.16
E
129.12
Q
128.14
W
186.21
F
147.18
T
101.11
E
129.12
D
115.09
P
97.12
G
57.05
P
97.12
D
115.09
E
129.12
A
71.09
P
97.12
R
156.19
M
131.19
P
97.12
E
129.12
A
71.09
A
71.09
P
97.12
P
97.12
V
99.14
A
71.09
P
97.12
A
71.09
P
97.12
A
71.09
A
71.09
P
97.12
T
101.11
P
97.12
A
71.09
A
71.09
P
97.12
A
71.09
P
97.12
A
71.09
P
97.12
S
87.08
W
186.21
P
97.12
L
113.16
S
87.08
S
87.08
S
87.08
V
99.14
P
97.12
S
87.08
Q
128.14
K
128.17
T
101.11
Y
163.18
Q
128.14
G
57.05
S
87.08
Y
163.18
G
57.05
F
147.18
R
156.19
L
113.16
G
57.05
F
147.18
L
113.16
H
137.14
S
87.08
G
57.05
T
101

이 단백질의 분자량을 계산해 보자. 
각각의 아미노산의 분자량을 딕셔너리로 저장하고...

In [172]:
MolecularWeight

43636.37000000006

In [166]:
MW = {'G':57.05,
      'A':71.09,
      'S':87.08,
      'T':101.11,
      'C':103.15,
      'V':99.14,
      'L':113.16,
      'I':113.16,
      'M':131.19,
      'P':97.12,
      'F':147.18,
      'Y':163.18,
      'W':186.21,
      'D':115.09,
      'E':129.12,
      'N':114.11,
      'Q':128.14,
      'H':137.14,
      'K':128.17,
      'R':156.19}

In [167]:
MW['A']

71.09

In [168]:
MW['K']

128.17

단백질 서열을 처음부터 한 글자씩 읽은 다음, 이것을 딕셔너리에서 찾는다. 
찾는 숫자를 다 더한다. 

In [13]:
for aa in seq:
    print(aa)

M
E
E
P
Q
S
D
P
S
V
E
P
P
L
S
Q
E
T
F
S
D
L
W
K
L
L
P
E
N
N
V
L
S
P
L
P
S
Q
A
M
D
D
L
M
L
S
P
D
D
I
E
Q
W
F
T
E
D
P
G
P
D
E
A
P
R
M
P
E
A
A
P
P
V
A
P
A
P
A
A
P
T
P
A
A
P
A
P
A
P
S
W
P
L
S
S
S
V
P
S
Q
K
T
Y
Q
G
S
Y
G
F
R
L
G
F
L
H
S
G
T
A
K
S
V
T
C
T
Y
S
P
A
L
N
K
M
F
C
Q
L
A
K
T
C
P
V
Q
L
W
V
D
S
T
P
P
P
G
T
R
V
R
A
M
A
I
Y
K
Q
S
Q
H
M
T
E
V
V
R
R
C
P
H
H
E
R
C
S
D
S
D
G
L
A
P
P
Q
H
L
I
R
V
E
G
N
L
R
V
E
Y
L
D
D
R
N
T
F
R
H
S
V
V
V
P
Y
E
P
P
E
V
G
S
D
C
T
T
I
H
Y
N
Y
M
C
N
S
S
C
M
G
G
M
N
R
R
P
I
L
T
I
I
T
L
E
D
S
S
G
N
L
L
G
R
N
S
F
E
V
R
V
C
A
C
P
G
R
D
R
R
T
E
E
E
N
L
R
K
K
G
E
P
H
H
E
L
P
P
G
S
T
K
R
A
L
P
N
N
T
S
S
S
P
Q
P
K
K
K
P
L
D
G
E
Y
F
T
L
Q
I
R
G
R
E
R
F
E
M
F
R
E
L
N
E
A
L
E
L
K
D
A
Q
A
G
K
E
P
G
G
S
R
A
H
S
S
H
L
K
S
K
K
G
Q
S
T
S
R
H
K
K
L
M
F
K
T
E
G
P
D
S
D


In [173]:
for aa in seq:
    print (MW[aa])

131.19
129.12
129.12
97.12
128.14
87.08
115.09
97.12
87.08
99.14
129.12
97.12
97.12
113.16
87.08
128.14
129.12
101.11
147.18
87.08
115.09
113.16
186.21
128.17
113.16
113.16
97.12
129.12
114.11
114.11
99.14
113.16
87.08
97.12
113.16
97.12
87.08
128.14
71.09
131.19
115.09
115.09
113.16
131.19
113.16
87.08
97.12
115.09
115.09
113.16
129.12
128.14
186.21
147.18
101.11
129.12
115.09
97.12
57.05
97.12
115.09
129.12
71.09
97.12
156.19
131.19
97.12
129.12
71.09
71.09
97.12
97.12
99.14
71.09
97.12
71.09
97.12
71.09
71.09
97.12
101.11
97.12
71.09
71.09
97.12
71.09
97.12
71.09
97.12
87.08
186.21
97.12
113.16
87.08
87.08
87.08
99.14
97.12
87.08
128.14
128.17
101.11
163.18
128.14
57.05
87.08
163.18
57.05
147.18
156.19
113.16
57.05
147.18
113.16
137.14
87.08
57.05
101.11
71.09
128.17
87.08
99.14
101.11
103.15
101.11
163.18
87.08
97.12
71.09
113.16
114.11
128.17
131.19
147.18
103.15
128.14
113.16
71.09
128.17
101.11
103.15
97.12
99.14
128.14
113.16
186.21
99.14
115.09
87.08
101.11
97.12
97.12
97.12
5

어떻게 이 숫자를 더할 것인가? 

In [175]:
molecular_weight=0
for aa in seq:
    molecular_weight=molecular_weight+MW[aa]
    
#
#   molecular_weight+=molecular_weight 와 같이 써도 된다 
#
print (molecular_weight)

43636.37000000006


In [15]:
molecular_weight

43636.37000000006

### 연습문제

각각의 아미노산과 분자량을 같이 표시해보자.  

Q 128.14

S 87.08

T 101.11

S 87.08

R 156.19

H 137.14

K 128.17

K 128.17

L 113.16

M 131.19

F 147.18

K 128.17

T 101.11

E 129.12

G 57.05

P 97.12

D 115.09

S 87.08

D 115.09

Total  43636.37000000006

이런 출력이 나오도록 위의 스크립트를 고쳐 보자. 

# 유닉스 명령 사용하기 

Jupyter notebook에서도 유닉스 명령을 내릴 수 있는데..
! (느낌표)를 치고 유닉스 명령을 치면 실행된다. 

In [177]:
!curl -O https://uniprot.org/uniprot/P59594.fasta 

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1369  100  1369    0     0   1198      0  0:00:01  0:00:01 --:--:--  1198


In [178]:
!cat P59594.fasta

>sp|P59594|SPIKE_CVHSA Spike glycoprotein OS=Human SARS coronavirus OX=694009 GN=S PE=1 SV=1
MFIFLLFLTLTSGSDLDRCTTFDDVQAPNYTQHTSSMRGVYYPDEIFRSDTLYLTQDLFL
PFYSNVTGFHTINHTFGNPVIPFKDGIYFAATEKSNVVRGWVFGSTMNNKSQSVIIINNS
TNVVIRACNFELCDNPFFAVSKPMGTQTHTMIFDNAFNCTFEYISDAFSLDVSEKSGNFK
HLREFVFKNKDGFLYVYKGYQPIDVVRDLPSGFNTLKPIFKLPLGINITNFRAILTAFSP
AQDIWGTSAAAYFVGYLKPTTFMLKYDENGTITDAVDCSQNPLAELKCSVKSFEIDKGIY
QTSNFRVVPSGDVVRFPNITNLCPFGEVFNATKFPSVYAWERKKISNCVADYSVLYNSTF
FSTFKCYGVSATKLNDLCFSNVYADSFVVKGDDVRQIAPGQTGVIADYNYKLPDDFMGCV
LAWNTRNIDATSTGNYNYKYRYLRHGKLRPFERDISNVPFSPDGKPCTPPALNCYWPLND
YGFYTTTGIGYQPYRVVVLSFELLNAPATVCGPKLSTDLIKNQCVNFNFNGLTGTGVLTP
SSKRFQPFQQFGRDVSDFTDSVRDPKTSEILDISPCSFGGVSVITPGTNASSEVAVLYQD
VNCTDVSTAIHADQLTPAWRIYSTGNNVFQTQAGCLIGAEHVDTSYECDIPIGAGICASY
HTVSLLRSTSQKSIVAYTMSLGADSSIAYSNNTIAIPTNFSISITTEVMPVSMAKTSVDC
NMYICGDSTECANLLLQYGSFCTQLNRALSGIAAEQDRNTREVFAQVKQMYKTPTLKYFG
GFNFSQILPDPLKPTKRSFIEDLLFNKVTLADAGFMKQYGECLGDINARDLICAQKFNGL
TVLPPLLTDDMIAAYTAALVSGTATAGWTFGAGAALQIPFAMQMAYRFNGIGV

In [8]:
!ls

Chapter4-2.ipynb Lect3.pptx       list.txt         ~$Lect3.pptx
Chapter4.ipynb   Lect4.pptx       pdbdownload
Lect1.pptx       P59594.fasta     ~$Lect1.pptx
Lect2.pptx       blalist          ~$Lect2.pptx


In [9]:
!cat P59594.fasta

>sp|P59594|SPIKE_CVHSA Spike glycoprotein OS=Human SARS coronavirus OX=694009 GN=S PE=1 SV=1
MFIFLLFLTLTSGSDLDRCTTFDDVQAPNYTQHTSSMRGVYYPDEIFRSDTLYLTQDLFL
PFYSNVTGFHTINHTFGNPVIPFKDGIYFAATEKSNVVRGWVFGSTMNNKSQSVIIINNS
TNVVIRACNFELCDNPFFAVSKPMGTQTHTMIFDNAFNCTFEYISDAFSLDVSEKSGNFK
HLREFVFKNKDGFLYVYKGYQPIDVVRDLPSGFNTLKPIFKLPLGINITNFRAILTAFSP
AQDIWGTSAAAYFVGYLKPTTFMLKYDENGTITDAVDCSQNPLAELKCSVKSFEIDKGIY
QTSNFRVVPSGDVVRFPNITNLCPFGEVFNATKFPSVYAWERKKISNCVADYSVLYNSTF
FSTFKCYGVSATKLNDLCFSNVYADSFVVKGDDVRQIAPGQTGVIADYNYKLPDDFMGCV
LAWNTRNIDATSTGNYNYKYRYLRHGKLRPFERDISNVPFSPDGKPCTPPALNCYWPLND
YGFYTTTGIGYQPYRVVVLSFELLNAPATVCGPKLSTDLIKNQCVNFNFNGLTGTGVLTP
SSKRFQPFQQFGRDVSDFTDSVRDPKTSEILDISPCSFGGVSVITPGTNASSEVAVLYQD
VNCTDVSTAIHADQLTPAWRIYSTGNNVFQTQAGCLIGAEHVDTSYECDIPIGAGICASY
HTVSLLRSTSQKSIVAYTMSLGADSSIAYSNNTIAIPTNFSISITTEVMPVSMAKTSVDC
NMYICGDSTECANLLLQYGSFCTQLNRALSGIAAEQDRNTREVFAQVKQMYKTPTLKYFG
GFNFSQILPDPLKPTKRSFIEDLLFNKVTLADAGFMKQYGECLGDINARDLICAQKFNGL
TVLPPLLTDDMIAAYTAALVSGTATAGWTFGAGAALQIPFAMQMAYRFNGIGV

In [179]:
!blastp -query P59594.fasta -db swissprot -out blastoutput -outfmt 6 -num_threads 16

In [180]:
!cat blastoutput

sp|P59594|SPIKE_CVHSA	P59594.1	100.000	1255	0	0	1	1255	1	1255	0.0	2602
sp|P59594|SPIKE_CVHSA	Q0Q475.1	79.064	1261	236	5	1	1255	3	1241	0.0	2062
sp|P59594|SPIKE_CVHSA	Q3I5J5.1	78.668	1261	243	4	1	1255	1	1241	0.0	2056
sp|P59594|SPIKE_CVHSA	Q3LZX1.1	78.509	1261	244	7	1	1255	3	1242	0.0	2050
sp|P59594|SPIKE_CVHSA	A3EXG6.1	32.479	1287	749	32	29	1248	34	1267	0.0	631
sp|P59594|SPIKE_CVHSA	A3EXD0.1	35.801	1148	648	36	152	1246	238	1349	0.0	590
sp|P59594|SPIKE_CVHSA	Q0Q4F2.1	36.608	1079	610	27	222	1246	289	1347	0.0	583
sp|P59594|SPIKE_CVHSA	A3EX94.1	36.506	1082	609	28	222	1246	289	1349	3.10e-180	571
sp|P59594|SPIKE_CVHSA	K9N5Q8.1	34.791	1052	615	28	250	1245	311	1347	2.45e-172	550
sp|P59594|SPIKE_CVHSA	P11225.1	31.008	1319	757	39	1	1253	2	1233	2.72e-164	525
sp|P59594|SPIKE_CVHSA	P11224.2	30.312	1346	785	41	1	1236	2	1304	7.14e-158	510
sp|P59594|SPIKE_CVHSA	P25192.1	37.071	758	434	12	514	1248	621	1358	9.88e-145	476
sp|P59594|SPIKE_CVHSA	P25192.1	24.678	466	275	18	1	424	1	432	1.37e-20	102
sp|P59594|SP

유닉스 명령어를 노트북을 이용하여 이용햐면 기록이 남으니 매우 편리! 

파이썬에서도 텍스트 파일 출력을 읽을 수 있다. 
앞에서 blast의 출력 파일인 blastoutput을 읽어보자.

파일을 읽기 위해서는 일단 파일을 지정해야 한다. 

In [181]:
!cat blastoutput

sp|P59594|SPIKE_CVHSA	P59594.1	100.000	1255	0	0	1	1255	1	1255	0.0	2602
sp|P59594|SPIKE_CVHSA	Q0Q475.1	79.064	1261	236	5	1	1255	3	1241	0.0	2062
sp|P59594|SPIKE_CVHSA	Q3I5J5.1	78.668	1261	243	4	1	1255	1	1241	0.0	2056
sp|P59594|SPIKE_CVHSA	Q3LZX1.1	78.509	1261	244	7	1	1255	3	1242	0.0	2050
sp|P59594|SPIKE_CVHSA	A3EXG6.1	32.479	1287	749	32	29	1248	34	1267	0.0	631
sp|P59594|SPIKE_CVHSA	A3EXD0.1	35.801	1148	648	36	152	1246	238	1349	0.0	590
sp|P59594|SPIKE_CVHSA	Q0Q4F2.1	36.608	1079	610	27	222	1246	289	1347	0.0	583
sp|P59594|SPIKE_CVHSA	A3EX94.1	36.506	1082	609	28	222	1246	289	1349	3.10e-180	571
sp|P59594|SPIKE_CVHSA	K9N5Q8.1	34.791	1052	615	28	250	1245	311	1347	2.45e-172	550
sp|P59594|SPIKE_CVHSA	P11225.1	31.008	1319	757	39	1	1253	2	1233	2.72e-164	525
sp|P59594|SPIKE_CVHSA	P11224.2	30.312	1346	785	41	1	1236	2	1304	7.14e-158	510
sp|P59594|SPIKE_CVHSA	P25192.1	37.071	758	434	12	514	1248	621	1358	9.88e-145	476
sp|P59594|SPIKE_CVHSA	P25192.1	24.678	466	275	18	1	424	1	432	1.37e-20	102
sp|P59594|SP

In [182]:
f=open('blastoutput','r')
#open 이라는 명령어를 사용한다
#파일 이름을 지정하고
#파일을 읽을 때는 'r', 파일을 기록할 때는 'w'를 이용한다. 
filecontent = f.readlines()
f.close()
#파일 내용을 몽땅 filecontent 라는 list 에 기록한다.

In [184]:
for line in filecontent:
    print (line)

sp|P59594|SPIKE_CVHSA	P59594.1	100.000	1255	0	0	1	1255	1	1255	0.0	2602

sp|P59594|SPIKE_CVHSA	Q0Q475.1	79.064	1261	236	5	1	1255	3	1241	0.0	2062

sp|P59594|SPIKE_CVHSA	Q3I5J5.1	78.668	1261	243	4	1	1255	1	1241	0.0	2056

sp|P59594|SPIKE_CVHSA	Q3LZX1.1	78.509	1261	244	7	1	1255	3	1242	0.0	2050

sp|P59594|SPIKE_CVHSA	A3EXG6.1	32.479	1287	749	32	29	1248	34	1267	0.0	631

sp|P59594|SPIKE_CVHSA	A3EXD0.1	35.801	1148	648	36	152	1246	238	1349	0.0	590

sp|P59594|SPIKE_CVHSA	Q0Q4F2.1	36.608	1079	610	27	222	1246	289	1347	0.0	583

sp|P59594|SPIKE_CVHSA	A3EX94.1	36.506	1082	609	28	222	1246	289	1349	3.10e-180	571

sp|P59594|SPIKE_CVHSA	K9N5Q8.1	34.791	1052	615	28	250	1245	311	1347	2.45e-172	550

sp|P59594|SPIKE_CVHSA	P11225.1	31.008	1319	757	39	1	1253	2	1233	2.72e-164	525

sp|P59594|SPIKE_CVHSA	P11224.2	30.312	1346	785	41	1	1236	2	1304	7.14e-158	510

sp|P59594|SPIKE_CVHSA	P25192.1	37.071	758	434	12	514	1248	621	1358	9.88e-145	476

sp|P59594|SPIKE_CVHSA	P25192.1	24.678	466	275	18	1	424	1	432	1.37e-20	102


In [4]:
#각각의 줄이 별도의 항목으로 리스트에 저장된다. 
#\t 는 탭 문자를 의미하며 \n는 줄 띄우기 문자를 의미한다. 
#읽어서 filecontent 에 저장된 내용을 보기 위해서는 

for line in filecontent:
    print(line)

sp|P59594|SPIKE_CVHSA	P59594.1	100.000	1255	0	0	1	1255	1	1255	0.0	2602

sp|P59594|SPIKE_CVHSA	Q0Q475.1	79.064	1261	236	5	1	1255	3	1241	0.0	2062

sp|P59594|SPIKE_CVHSA	Q3I5J5.1	78.668	1261	243	4	1	1255	1	1241	0.0	2056

sp|P59594|SPIKE_CVHSA	Q3LZX1.1	78.509	1261	244	7	1	1255	3	1242	0.0	2050

sp|P59594|SPIKE_CVHSA	A3EXG6.1	32.479	1287	749	32	29	1248	34	1267	0.0	631

sp|P59594|SPIKE_CVHSA	A3EXD0.1	35.801	1148	648	36	152	1246	238	1349	0.0	590

sp|P59594|SPIKE_CVHSA	Q0Q4F2.1	36.608	1079	610	27	222	1246	289	1347	0.0	583

sp|P59594|SPIKE_CVHSA	A3EX94.1	36.506	1082	609	28	222	1246	289	1349	3.10e-180	571

sp|P59594|SPIKE_CVHSA	K9N5Q8.1	34.791	1052	615	28	250	1245	311	1347	2.45e-172	550

sp|P59594|SPIKE_CVHSA	P11225.1	31.008	1319	757	39	1	1253	2	1233	2.72e-164	525

sp|P59594|SPIKE_CVHSA	P11224.2	30.312	1346	785	41	1	1236	2	1304	7.14e-158	510

sp|P59594|SPIKE_CVHSA	P25192.1	37.071	758	434	12	514	1248	621	1358	9.88e-145	476

sp|P59594|SPIKE_CVHSA	P25192.1	24.678	466	275	18	1	424	1	432	1.37e-20	102


이렇게 하면.

읽어내는 줄을 '\t' (탭 문자) 기준으로 분리하여 

linecontent 라는 리스트에 저장한다. 

line.split('\t') <- 탭 문자 

In [187]:
for line in filecontent:
    linecontent = line.split('\t')
    print(linecontent[1])

P59594.1
Q0Q475.1
Q3I5J5.1
Q3LZX1.1
A3EXG6.1
A3EXD0.1
Q0Q4F2.1
A3EX94.1
K9N5Q8.1
P11225.1
P11224.2
P25192.1
P25192.1
Q9QAQ8.1
Q9QAQ8.1
Q91A26.1
Q91A26.1
Q9QAR5.1
Q9QAR5.1
Q8V436.1
Q8V436.1
P25190.1
P25190.1
P25194.1
P25194.1
P25191.1
P25191.1
P15777.1
P15777.1
P25193.2
P25193.2
Q0ZME7.1
Q0ZME7.1
P36334.1
P36334.1
Q14EB0.1
Q14EB0.1
Q8BB25.1
Q8BB25.1
Q9IKD1.1
Q9IKD1.1
Q02385.1
Q02385.1
P22432.1
P22432.1
Q8JSP8.1
Q8JSP8.1
Q5MQD0.1
Q5MQD0.1
P15423.1
P12650.1
P12651.1
P05135.1
P12722.1
P11223.1
Q0Q466.1
P24413.1
P27655.1
Q65984.1
Q6Q1S2.1
P10033.1
P07946.2
P33470.2
Q01977.1
Q7T6T3.1
P18450.2
P36300.1
Q91AV1.1
B8F3P6.1
A5UEK6.1
Q7VL94.1
Q38WX8.1
P45185.2
A5UBY0.1
O30867.1
B7UN08.1
B5R8K9.1
P60107.1
Q2YY12.1
Q2FH32.1
A8AIF7.1
Q6G9F5.1
Q8N9W4.3
B4T156.1
Q8Z7Z7.1
A9N7T9.1
A7MZI2.1
P48479.2
A9MHW1.1
B5YT56.1
A7ZK12.1
B7LN84.1
O67068.1
A7ZYM4.1
A1A9J9.1
P75434.1
Q31YS8.1
A1S7B2.1
B2TUF7.1
P0DJF6.1
B1IW09.1
Q6D445.1
A3MZU5.1
P37535.1
Q9UL45.1
B0BUD7.1
B3GXC3.1
A8H5M2.1
B0TQ01.1


In [188]:
for line in filecontent:
    linecontent = line.split('\t')
    print(linecontent[0])

sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|SPIKE_CVHSA
sp|P59594|

P59594.1...만을 출력하려면 어떻게 할까? 

In [189]:
for line in filecontent:
    linecontent = line.split('\t')
    print(linecontent[1])

P59594.1
Q0Q475.1
Q3I5J5.1
Q3LZX1.1
A3EXG6.1
A3EXD0.1
Q0Q4F2.1
A3EX94.1
K9N5Q8.1
P11225.1
P11224.2
P25192.1
P25192.1
Q9QAQ8.1
Q9QAQ8.1
Q91A26.1
Q91A26.1
Q9QAR5.1
Q9QAR5.1
Q8V436.1
Q8V436.1
P25190.1
P25190.1
P25194.1
P25194.1
P25191.1
P25191.1
P15777.1
P15777.1
P25193.2
P25193.2
Q0ZME7.1
Q0ZME7.1
P36334.1
P36334.1
Q14EB0.1
Q14EB0.1
Q8BB25.1
Q8BB25.1
Q9IKD1.1
Q9IKD1.1
Q02385.1
Q02385.1
P22432.1
P22432.1
Q8JSP8.1
Q8JSP8.1
Q5MQD0.1
Q5MQD0.1
P15423.1
P12650.1
P12651.1
P05135.1
P12722.1
P11223.1
Q0Q466.1
P24413.1
P27655.1
Q65984.1
Q6Q1S2.1
P10033.1
P07946.2
P33470.2
Q01977.1
Q7T6T3.1
P18450.2
P36300.1
Q91AV1.1
B8F3P6.1
A5UEK6.1
Q7VL94.1
Q38WX8.1
P45185.2
A5UBY0.1
O30867.1
B7UN08.1
B5R8K9.1
P60107.1
Q2YY12.1
Q2FH32.1
A8AIF7.1
Q6G9F5.1
Q8N9W4.3
B4T156.1
Q8Z7Z7.1
A9N7T9.1
A7MZI2.1
P48479.2
A9MHW1.1
B5YT56.1
A7ZK12.1
B7LN84.1
O67068.1
A7ZYM4.1
A1A9J9.1
P75434.1
Q31YS8.1
A1S7B2.1
B2TUF7.1
P0DJF6.1
B1IW09.1
Q6D445.1
A3MZU5.1
P37535.1
Q9UL45.1
B0BUD7.1
B3GXC3.1
A8H5M2.1
B0TQ01.1


화면에 출력하지 말고 리스트에 저장하려면 어떻게 할까? 


In [190]:
#swissprot id 를 저장할 새로운 리스트 
swissprot = []

for line in filecontent:
    linecontent = line.split('\t')
    swissprot.append(linecontent[1])

In [191]:
swissprot

['P59594.1',
 'Q0Q475.1',
 'Q3I5J5.1',
 'Q3LZX1.1',
 'A3EXG6.1',
 'A3EXD0.1',
 'Q0Q4F2.1',
 'A3EX94.1',
 'K9N5Q8.1',
 'P11225.1',
 'P11224.2',
 'P25192.1',
 'P25192.1',
 'Q9QAQ8.1',
 'Q9QAQ8.1',
 'Q91A26.1',
 'Q91A26.1',
 'Q9QAR5.1',
 'Q9QAR5.1',
 'Q8V436.1',
 'Q8V436.1',
 'P25190.1',
 'P25190.1',
 'P25194.1',
 'P25194.1',
 'P25191.1',
 'P25191.1',
 'P15777.1',
 'P15777.1',
 'P25193.2',
 'P25193.2',
 'Q0ZME7.1',
 'Q0ZME7.1',
 'P36334.1',
 'P36334.1',
 'Q14EB0.1',
 'Q14EB0.1',
 'Q8BB25.1',
 'Q8BB25.1',
 'Q9IKD1.1',
 'Q9IKD1.1',
 'Q02385.1',
 'Q02385.1',
 'P22432.1',
 'P22432.1',
 'Q8JSP8.1',
 'Q8JSP8.1',
 'Q5MQD0.1',
 'Q5MQD0.1',
 'P15423.1',
 'P12650.1',
 'P12651.1',
 'P05135.1',
 'P12722.1',
 'P11223.1',
 'Q0Q466.1',
 'P24413.1',
 'P27655.1',
 'Q65984.1',
 'Q6Q1S2.1',
 'P10033.1',
 'P07946.2',
 'P33470.2',
 'Q01977.1',
 'Q7T6T3.1',
 'P18450.2',
 'P36300.1',
 'Q91AV1.1',
 'B8F3P6.1',
 'A5UEK6.1',
 'Q7VL94.1',
 'Q38WX8.1',
 'P45185.2',
 'A5UBY0.1',
 'O30867.1',
 'B7UN08.1',
 'B5R8K9.1',

이것을 파일에 저장해보려고 한다. 어떻게 해야 할까? 

In [192]:
f=open('swissoutput','w')
#open 이라는 명령어를 사용한다
#파일 이름을 지정하고
#파일을 기록할 때는 'w'를 이용한다. 
f.writelines(swissprot)
f.close()

In [194]:
!cat swissoutput

P59594.1Q0Q475.1Q3I5J5.1Q3LZX1.1A3EXG6.1A3EXD0.1Q0Q4F2.1A3EX94.1K9N5Q8.1P11225.1P11224.2P25192.1P25192.1Q9QAQ8.1Q9QAQ8.1Q91A26.1Q91A26.1Q9QAR5.1Q9QAR5.1Q8V436.1Q8V436.1P25190.1P25190.1P25194.1P25194.1P25191.1P25191.1P15777.1P15777.1P25193.2P25193.2Q0ZME7.1Q0ZME7.1P36334.1P36334.1Q14EB0.1Q14EB0.1Q8BB25.1Q8BB25.1Q9IKD1.1Q9IKD1.1Q02385.1Q02385.1P22432.1P22432.1Q8JSP8.1Q8JSP8.1Q5MQD0.1Q5MQD0.1P15423.1P12650.1P12651.1P05135.1P12722.1P11223.1Q0Q466.1P24413.1P27655.1Q65984.1Q6Q1S2.1P10033.1P07946.2P33470.2Q01977.1Q7T6T3.1P18450.2P36300.1Q91AV1.1B8F3P6.1A5UEK6.1Q7VL94.1Q38WX8.1P45185.2A5UBY0.1O30867.1B7UN08.1B5R8K9.1P60107.1Q2YY12.1Q2FH32.1A8AIF7.1Q6G9F5.1Q8N9W4.3B4T156.1Q8Z7Z7.1A9N7T9.1A7MZI2.1P48479.2A9MHW1.1B5YT56.1A7ZK12.1B7LN84.1O67068.1A7ZYM4.1A1A9J9.1P75434.1Q31YS8.1A1S7B2.1B2TUF7.1P0DJF6.1B1IW09.1Q6D445.1A3MZU5.1P37535.1Q9UL45.1B0BUD7.1B3GXC3.1A8H5M2.1B0TQ01.1

줄 간 구분이 안 되어 있다. 이것을 고치려면 어떻게 할까? 

텍스트 파일에서 쓰기를 하려면 각각에 새로 줄 띄기 문자 ('\n')를 넣어야 한다. 이를 위해서 다음과 같이 수정하자. 

In [195]:
swissprot = []

for line in filecontent:
    linecontent = line.split('\t')
    #파일 줄 뒤에 '\n' 문자를 추가하였다. 
    swissprot.append(linecontent[1]+'\n')
f=open('swissoutput','w')
f.writelines(swissprot)
f.close()

In [196]:
!cat swissoutput

P59594.1
Q0Q475.1
Q3I5J5.1
Q3LZX1.1
A3EXG6.1
A3EXD0.1
Q0Q4F2.1
A3EX94.1
K9N5Q8.1
P11225.1
P11224.2
P25192.1
P25192.1
Q9QAQ8.1
Q9QAQ8.1
Q91A26.1
Q91A26.1
Q9QAR5.1
Q9QAR5.1
Q8V436.1
Q8V436.1
P25190.1
P25190.1
P25194.1
P25194.1
P25191.1
P25191.1
P15777.1
P15777.1
P25193.2
P25193.2
Q0ZME7.1
Q0ZME7.1
P36334.1
P36334.1
Q14EB0.1
Q14EB0.1
Q8BB25.1
Q8BB25.1
Q9IKD1.1
Q9IKD1.1
Q02385.1
Q02385.1
P22432.1
P22432.1
Q8JSP8.1
Q8JSP8.1
Q5MQD0.1
Q5MQD0.1
P15423.1
P12650.1
P12651.1
P05135.1
P12722.1
P11223.1
Q0Q466.1
P24413.1
P27655.1
Q65984.1
Q6Q1S2.1
P10033.1
P07946.2
P33470.2
Q01977.1
Q7T6T3.1
P18450.2
P36300.1
Q91AV1.1
B8F3P6.1
A5UEK6.1
Q7VL94.1
Q38WX8.1
P45185.2
A5UBY0.1
O30867.1
B7UN08.1
B5R8K9.1
P60107.1
Q2YY12.1
Q2FH32.1
A8AIF7.1
Q6G9F5.1
Q8N9W4.3
B4T156.1
Q8Z7Z7.1
A9N7T9.1
A7MZI2.1
P48479.2
A9MHW1.1
B5YT56.1
A7ZK12.1
B7LN84.1
O67068.1
A7ZYM4.1
A1A9J9.1
P75434.1
Q31YS8.1
A1S7B2.1
B2TUF7.1
P0DJF6.1
B1IW09.1
Q6D445.1
A3MZU5.1
P37535.1
Q9UL45.1
B0BUD7.1
B3GXC3.1
A8H5M2.1
B0TQ01.1


결국 유닉스 명령어로 cat blastoutput.txt | cut -d$'\t' -f 2 > swissoutput 하는 것과 동일한 일이지만, 파이썬으로 수행 가능하다. 
